From a557508590dca1536345e13ba1b043f33defe13a Mon Sep 17 00:00:00 2001
Patch-Source: https://github.com/dotnet/runtime/pull/62942
From: Antoine Martin <dev@ayakael.net>
Date: Wed, 1 Dec 2021 02:52:05 +0000
Subject: [PATCH] Fix last version digit present on alpine non-portable

Runtime adds the extra subversion in its calculation of Alpine's
DistroRID when a non-portable build, but does so inconsistently.
This creates an error when it generates its nuget package.

---
 eng/native/init-distro-rid.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eng/native/init-distro-rid.sh b/eng/native/init-distro-rid.sh
index b6921a90346..091738944a7 100644
--- a/eng/native/init-distro-rid.sh
+++ b/eng/native/init-distro-rid.sh
@@ -41,7 +41,7 @@ initNonPortableDistroRid()
             # We have forced __PortableBuild=0. This is because -portablebuld
             # has been passed as false.
             if (( isPortable == 0 )); then
-                if [ "${ID}" = "rhel" ]; then
+                if [ "${ID}" = "rhel" ] || [ "${ID}" = "alpine" ]; then
                     # remove the last version digit
                     VERSION_ID="${VERSION_ID%.*}"
                 fi
-- 
2.34.0

